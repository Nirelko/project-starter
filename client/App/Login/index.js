import React from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import * as LoginActions from './redux';
import Login from './Login';

const LoginContainer = ({localLogin}) => (
  <Login onSubmit={localLogin} />
);

const mapStateToProps = state => state;

const mapDispatchToProps = dispatch => bindActionCreators(LoginActions, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LoginContainer);
