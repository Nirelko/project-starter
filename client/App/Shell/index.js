import React from 'react';
import {grey900} from 'material-ui/styles/colors';
import {AppBar, Card} from 'material-ui';

const style = {
  wrapper: {
    height: '100%',
    backgroundColor: grey900,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
  },
  card: {
    width: '80%'
  }
};

export default ({children}) => {
  return (
    <div style={style.wrapper}>
      <Card style={style.card}>
        <AppBar title='Welcome to the project-starter' showMenuIconButton={false} />
        {children}
      </Card>
    </div>
  );
};