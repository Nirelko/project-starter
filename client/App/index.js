import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {blue500} from 'material-ui/styles/colors';
import {Flex} from 'reflexbox';

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: blue500
  }
});

export default ({children}) => {
  let devTools = null;

  if (process.env.NODE_ENV !== 'production' && !window.devToolsExtension) {
    const DevTools = require('../components/DevTools').default;

    devTools = <DevTools />;
  }

  return (
    <MuiThemeProvider muiTheme={muiTheme}>
      <Flex auto>
        {children}
        {devTools}
      </Flex>
    </MuiThemeProvider>
  );
};