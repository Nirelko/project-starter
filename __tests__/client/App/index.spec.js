import React from 'react';
import App from '../../../client/App/index';
import renderer from 'react-test-renderer';

describe('App component', () => {
  it('should render properly', () => {
    const component = renderer.create(
      <App />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
