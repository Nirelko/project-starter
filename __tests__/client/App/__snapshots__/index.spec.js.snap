// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render properly 1`] = `
<div
  className="_Rfx0_ _Rfx1_"
/>
`;
